src/lib/state/network/NetworkSimulationEvent.ts

```diff
@@ -0,0 +1,29 @@
+import Signal, { Connection } from "../../utilities/simple-signal";
+import { NetworkEvent } from "./NetworkEvent";
+
+const SimulatedPlayer = { Name: "NetworkSimulation" } as Player;
+
+export class NetworkSimulationEvent extends NetworkEvent {
+	private serverSignal = new Signal<[Player, ...Array<unknown>]>();
+	private clientSignal = new Signal<Array<unknown>>();
+
+	public FireClient(_player: Player, ...args: Array<unknown>): void {
+		this.clientSignal.Fire(...args);
+	}
```

Comment:
Pretty sure we want to keep all class-related methods to be `camelCase`. Also, I might as well hop on this and rework the Internal Networking handle, as I mentioned some weeks ago.